ghost/core/core/server/services/stats/TopPostsStatsService.js

```
@@ -0,0 +1,196 @@
+     * @param {TopPostsOptions} options
+     * @returns {Promise<{data: TopPostResult[]}>} The top posts based on the requested attribution metric
+     */
+    async getTopPosts(options = {}) {
+        logging.info('TopPostsStatsService.getTopPosts called with options:', options);
+        try {
+            const order = options.order || 'free_members desc';
+            const limit = options.limit || 20;
+            const [orderField, orderDirection = 'desc'] = order.split(' ');
+
+            if (!['free_members', 'paid_members', 'mrr'].includes(orderField)) {
+                throw new Error(`Invalid order field: ${orderField}. Must be one of: free_members, paid_members, mrr`);
```

ESLint

[error] Direct calls to new Error() are not allowed. Please use @tryghost/errors.

(ghost/ghost-custom/no-native-error)